ioservice/io_foms.c

```diff
@@ -2123,20 +2153,20 @@ static int m0_io_fom_cob_rw_tick(struct m0_fom *fom)
 
 	M0_ENTRY("fom %p, fop %p, item %p[%u] %s" FID_F, fom, fom->fo_fop,
 		 m0_fop_to_rpc_item(fom->fo_fop), m0_fop_opcode(fom->fo_fop),
-		 m0_fom_phase_name(fom, m0_fom_phase(fom)),
+		 m0_fom_phase_name(fom, phase),
 		 FID_P(&rwfop->crw_fid));
 
 	/* first handle generic phase */
-	if (m0_fom_phase(fom) == M0_FOPH_INIT) {
+	if (phase == M0_FOPH_INIT) {
```

Comment:
This whole block of code, from this LINE 2130, to  LINE 2184, should be re-organized.
It handles generic phases, but divided to three if conditions.
We can make it more readable.
We can refer to cob_ops_fom_tick().

Reply:
This can be an improvement in the future.